madlibs.js

```diff
@@ -22,21 +22,251 @@
  *  { word: "store", pos: "noun" }
  *  { word: "," }
  *  ....
- * 
+ *
  * There are multiple ways to do this, but you may want to use regular expressions.
  * Please go through this lesson: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/
  */
+
+/**
+This code is for a mad libs generator. 
+ It defines the input and output elements, and the arrays for these elements. 
+ The function parseStory takes in a string of a raw story and splits it into separate words. 
+The function then loops through each word and tests if it follows a pattern of having a word 
+followed by a square bracket with a single letter inside (either 'n', 'v', or 'a'). If the test is successful, 
+ the word is added to an object array with its corresponding part of speech (either 'noun', 'verb', or 'adjective').
+  If the test fails, the word is added to the object array without a part of speech. Finally, the parsed object array is returned.
+ */
+
+let inputElements,
+  outputElements,
+  inputElementsArray,
+  outputElementsArray = "";
+let edit = document.getElementById("madLibsEdit");
+let result = document.getElementById("madLibsPreview");
+
+// Function to parse a raw story into an object array
+// where each object represents a word in the story and its part of speech if applicable
+
 function parseStory(rawStory) {
-  // Your code here.
-  return {}; // This line is currently wrong :)
+  let obj = [];
+  // Split the raw story into individual words
+
+  let properties = rawStory.split(" ");
+  // Loop through each word
+
+  properties.forEach(function (singleword) {
+    // Test if the word follows the pattern of having a word followed by
+    // a square bracket with a single letter inside (either 'n', 'v', or 'a')
+    if (/(\S+)\[(.*?)\]/g.test(singleword)) {
```

Comment:
Good job on this. A switch case could have made it cleaner. You can check [`match()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) instead of `test` as well for regex.